src/main/java/com/minecolonies/core/colony/buildings/workerbuildings/BuildingConcreteMixer.java

```diff
@@ -184,18 +192,18 @@ public BlockPos getBlockToMine()
     @Nullable
     public BlockPos getBlockToPlace()
     {
-        for (int i = 1; i <= minWaterLevel; i++)
+        for (int i = 1; i <= WATER_DEPTH_SUPPORT; i++)
         {
             for (final BlockPos pos : waterPos.getOrDefault(i, Collections.emptyList()))
             {
                 final BlockState state = colony.getWorld().getBlockState(pos);
-                if (!state.getFluidState().isEmpty() && state.getBlock() == Blocks.WATER)
+                if (state.is(Blocks.WATER))
```

Comment:
flowing water is not water though

Reply:
Well I tested this, and it is.

If a block contains flowing water the block is still marked as containing water, the fluid state however just lists a different water depth.
There's no need for a depth check because the waterPos already has determined the required water level should be good in the schematic.

Reply:
ping

Reply:
Even if that is the case with low level water, then it's a non issue, because we only support up to a max depth of 5, and it worked correctly for every depth we support.

Reply:
For reference I had a debugger out, and every level it's blockstate was set to being the actual block of water, disregarding how high the water actually was

Reply:
I've double checked it again and it works fine. On top of that we already calculate the actual positions during construction, so I need not worry about calculating the depth of the water because the position itself is calculated to have water in it.
So that means we know the position is valid, so as long as there's then some water in it again, we're fine.